import sys
import os
import subprocess
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWidgets import QApplication, QFileDialog, QDialog, QPlainTextEdit
from PyQt5.QtWidgets import QMainWindow
import inspect

class Ui_moduleWeldPath(object):
	def setupUi(self, moduleWeldPath):
		moduleWeldPath.setObjectName("moduleWeldPath")
		moduleWeldPath.resize(822, 768)
		self.centralWidget = QtWidgets.QWidget(moduleWeldPath)
		self.centralWidget.setObjectName("centralWidget")
		
		#ds/dt
		self.v_t_header = QtWidgets.QLabel(self.centralWidget)
		self.v_t_header.setGeometry(QtCore.QRect(20, 10, 121, 18))
		self.v_t_header.setObjectName("v_t_header")
		self.v_t_table = QtWidgets.QTableWidget(self.centralWidget)
		self.v_t_table.setGeometry(QtCore.QRect(20, 30, 341, 161))
		self.v_t_table.setObjectName("v_t_table")
		self.v_t_table.setColumnCount(2)
		self.v_t_table.setRowCount(0)
		item = QtWidgets.QTableWidgetItem()
		self.v_t_table.setHorizontalHeaderItem(0, item)
		self.v_t_table.setColumnWidth(0,140)
		item = QtWidgets.QTableWidgetItem()
		self.v_t_table.setHorizontalHeaderItem(1, item)
		self.v_t_table.setColumnWidth(1,160)
		self.v_t_table.horizontalHeader().setCascadingSectionResizes(True)
		self.v_t_addRow_button = QtWidgets.QPushButton(self.centralWidget)
		self.v_t_addRow_button.setGeometry(QtCore.QRect(20, 200, 95, 26))
		self.v_t_addRow_button.setObjectName("v_t_addRow_button")
		#self.v_t_addRow_button.clicked.connect(self.addRow1)
		self.v_t_removeRow_button = QtWidgets.QPushButton(self.centralWidget)
		self.v_t_removeRow_button.setGeometry(QtCore.QRect(130, 200, 95, 26))
		self.v_t_removeRow_button.setObjectName("v_t_removeRow_button")
		#self.v_t_removeRow_button.clicked.connect(self.removeRow1)
		self.v_t_plot = QtWidgets.QLabel(self.centralWidget)
		self.v_t_plot.setGeometry(QtCore.QRect(560, 130, 231, 101))
		self.v_t_plot.setText("")
		self.v_t_plot.setPixmap(QtGui.QPixmap("Velocity_profile.png"))
		self.v_t_plot.setScaledContents(True)
		self.v_t_plot.setObjectName("v_t_plot")
		
		self.split1 = QtWidgets.QFrame(self.centralWidget)
		self.split1.setGeometry(QtCore.QRect(20, 230, 771, 16))
		self.split1.setFrameShape(QtWidgets.QFrame.HLine)
		self.split1.setFrameShadow(QtWidgets.QFrame.Sunken)
		self.split1.setObjectName("split1")
		
		#beam
		self.beam_header = QtWidgets.QLabel(self.centralWidget)
		self.beam_header.setGeometry(QtCore.QRect(20, 250, 121, 18))
		self.beam_header.setObjectName("beam_header")
		self.beam_t = QtWidgets.QLabel(self.centralWidget)
		self.beam_t.setGeometry(QtCore.QRect(20, 280, 71, 18))
		self.beam_t.setObjectName("beam_t")
		self.beam_on = QtWidgets.QTextEdit(self.centralWidget)
		self.beam_on.setGeometry(QtCore.QRect(70, 310, 91, 31))
		self.beam_on.setObjectName("beam_on")
		self.beam_on_label = QtWidgets.QLabel(self.centralWidget)
		self.beam_on_label.setGeometry(QtCore.QRect(20, 320, 51, 18))
		self.beam_on_label.setObjectName("beam_on_label")
		self.beam_off = QtWidgets.QTextEdit(self.centralWidget)
		self.beam_off.setGeometry(QtCore.QRect(250, 310, 91, 31))
		self.beam_off.setObjectName("beam_off")
		self.beam_off_label = QtWidgets.QLabel(self.centralWidget)
		self.beam_off_label.setGeometry(QtCore.QRect(200, 320, 51, 18))
		self.beam_off_label.setObjectName("beam_off_label")
		self.beam_plot = QtWidgets.QLabel(self.centralWidget)
		self.beam_plot.setGeometry(QtCore.QRect(560, 250, 231, 101))
		self.beam_plot.setText("")
		self.beam_plot.setPixmap(QtGui.QPixmap("Beam_profile.png"))
		self.beam_plot.setScaledContents(True)
		self.beam_plot.setObjectName("beam_plot")
		
		self.split2 = QtWidgets.QFrame(self.centralWidget)
		self.split2.setGeometry(QtCore.QRect(20, 360, 771, 16))
		self.split2.setFrameShape(QtWidgets.QFrame.HLine)
		self.split2.setFrameShadow(QtWidgets.QFrame.Sunken)
		self.split2.setObjectName("split2")
		
		#path
		self.path_header = QtWidgets.QLabel(self.centralWidget)
		self.path_header.setGeometry(QtCore.QRect(20, 380, 91, 18))
		self.path_header.setObjectName("path_header")
		self.path_table = QtWidgets.QTableWidget(self.centralWidget)
		self.path_table.setGeometry(QtCore.QRect(20, 400, 521, 192))
		self.path_table.setObjectName("path_table")
		self.path_table.setColumnCount(6)
		self.path_table.setRowCount(0)
		item = QtWidgets.QTableWidgetItem()
		self.path_table.setHorizontalHeaderItem(0, item)
		item = QtWidgets.QTableWidgetItem()
		self.path_table.setHorizontalHeaderItem(1, item)
		item = QtWidgets.QTableWidgetItem()
		self.path_table.setHorizontalHeaderItem(2, item)
		item = QtWidgets.QTableWidgetItem()
		self.path_table.setHorizontalHeaderItem(3, item)
		item = QtWidgets.QTableWidgetItem()
		self.path_table.setHorizontalHeaderItem(4, item)
		item = QtWidgets.QTableWidgetItem()
		self.path_table.setHorizontalHeaderItem(5, item)
		self.path_table.horizontalHeader().setDefaultSectionSize(80)
		self.path_addRow_button = QtWidgets.QPushButton(self.centralWidget)
		self.path_addRow_button.setGeometry(QtCore.QRect(20, 600, 95, 26))
		self.path_addRow_button.setObjectName("path_addRow_button")
		#self.path_addRow_button.clicked.connect(self.addRow2)
		self.path_removeRow_button = QtWidgets.QPushButton(self.centralWidget)
		self.path_removeRow_button.setGeometry(QtCore.QRect(130, 600, 95, 26))
		self.path_removeRow_button.setObjectName("path_removeRow_button")
		#self.path_removeRow_button.clicked.connect(self.removeRow2)
		self.path_plot = QtWidgets.QLabel(self.centralWidget)
		self.path_plot.setGeometry(QtCore.QRect(560, 410, 241, 171))
		self.path_plot.setText("")
		self.path_plot.setPixmap(QtGui.QPixmap("path_description.png"))
		self.path_plot.setScaledContents(True)
		self.path_plot.setObjectName("path_plot")
		
		self.dir0_label = QtWidgets.QLabel(self.centralWidget)
		self.dir0_label.setGeometry(QtCore.QRect(20, 660, 201, 18))
		self.dir0_label.setObjectName("dir0_label")
		self.dirx0 = QtWidgets.QTextEdit(self.centralWidget)
		self.dirx0.setGeometry(QtCore.QRect(230, 650, 31, 31))
		self.dirx0.setObjectName("dirx0")
		self.diry0 = QtWidgets.QTextEdit(self.centralWidget)
		self.diry0.setGeometry(QtCore.QRect(300, 650, 31, 31))
		self.diry0.setObjectName("diry0")
		self.dirz0 = QtWidgets.QTextEdit(self.centralWidget)
		self.dirz0.setGeometry(QtCore.QRect(370, 650, 31, 31))
		self.dirz0.setObjectName("dirz0")
		self.dirx0_label = QtWidgets.QLabel(self.centralWidget)
		self.dirx0_label.setGeometry(QtCore.QRect(210, 660, 21, 18))
		self.dirx0_label.setObjectName("dirx0_label")
		self.diry0_label = QtWidgets.QLabel(self.centralWidget)
		self.diry0_label.setGeometry(QtCore.QRect(280, 660, 21, 18))
		self.diry0_label.setObjectName("diry0_label")
		self.dirz0_label = QtWidgets.QLabel(self.centralWidget)
		self.dirz0_label.setGeometry(QtCore.QRect(350, 660, 21, 18))
		self.dirz0_label.setObjectName("dirz0_label")
		
		#self.button_apply = QtWidgets.QPushButton(self.centralWidget)
		#self.button_apply.setGeometry(QtCore.QRect(600, 610, 95, 26))
		#self.button_apply.setObjectName("button_apply")
		#self.button_apply.clicked.connect(self.apply)
		self.button_ok = QtWidgets.QPushButton(self.centralWidget)
		self.button_ok.setGeometry(QtCore.QRect(710, 670, 95, 26))
		self.button_ok.setObjectName("button_ok")
		#self.button_ok.clicked.connect(self.ok)
		
		moduleWeldPath.setCentralWidget(self.centralWidget)
		self.menuBar = QtWidgets.QMenuBar(moduleWeldPath)
		self.menuBar.setGeometry(QtCore.QRect(0, 0, 822, 23))
		self.menuBar.setObjectName("menuBar")
		moduleWeldPath.setMenuBar(self.menuBar)
		self.mainToolBar = QtWidgets.QToolBar(moduleWeldPath)
		self.mainToolBar.setObjectName("mainToolBar")
		moduleWeldPath.addToolBar(QtCore.Qt.TopToolBarArea, self.mainToolBar)
		self.statusBar = QtWidgets.QStatusBar(moduleWeldPath)
		self.statusBar.setObjectName("statusBar")
		moduleWeldPath.setStatusBar(self.statusBar)

		self.retranslateUi(moduleWeldPath)
		QtCore.QMetaObject.connectSlotsByName(moduleWeldPath)
		
	def retranslateUi(self, moduleWeldPath):
		_translate = QtCore.QCoreApplication.translate
		moduleWeldPath.setWindowTitle(_translate("moduleWeldPath", "moduleWeldPath"))
		self.v_t_header.setText(_translate("moduleWeldPath", "Velcoity Profile"))
		item = self.v_t_table.horizontalHeaderItem(0)
		item.setText(_translate("moduleWeldPath", "Time (s)"))
		item = self.v_t_table.horizontalHeaderItem(1)
		item.setText(_translate("moduleWeldPath", "Velocity (mm/s)"))
		self.v_t_addRow_button.setText(_translate("moduleWeldPath", "add"))
		self.v_t_removeRow_button.setText(_translate("moduleWeldPath", "remove"))
		
		self.beam_header.setText(_translate("moduleWeldPath", "Beam Profile"))
		self.beam_t.setText(_translate("moduleWeldPath", "Time (s):"))
		self.beam_on_label.setText(_translate("moduleWeldPath", "On (A)"))
		self.beam_off_label.setText(_translate("moduleWeldPath", "Off (B)"))
		
		self.path_header.setText(_translate("moduleWeldPath", "Torch Path"))
		item = self.path_table.horizontalHeaderItem(0)
		item.setText(_translate("moduleWeldPath", "x"))
		item = self.path_table.horizontalHeaderItem(1)
		item.setText(_translate("moduleWeldPath", "y"))
		item = self.path_table.horizontalHeaderItem(2)
		item.setText(_translate("moduleWeldPath", "z"))
		item = self.path_table.horizontalHeaderItem(3)
		item.setText(_translate("moduleWeldPath", "nx"))
		item = self.path_table.horizontalHeaderItem(4)
		item.setText(_translate("moduleWeldPath", "ny"))
		item = self.path_table.horizontalHeaderItem(5)
		item.setText(_translate("moduleWeldPath", "nz"))
		self.path_addRow_button.setText(_translate("moduleWeldPath", "add"))
		self.path_removeRow_button.setText(_translate("moduleWeldPath", "remove"))
		self.dir0_label.setText(_translate("moduleWeldPath", "Initial direction vector:"))
		self.dirx0_label.setText(_translate("moduleWeldPath", "x0"))
		self.diry0_label.setText(_translate("moduleWeldPath", "y0"))
		self.dirz0_label.setText(_translate("moduleWeldPath", "z0"))
		
		self.button_ok.setText(_translate("moduleWeldPath", "Ok"))
		#self.button_apply.setText(_translate("moduleWeldPath", "Apply"))
		
	
if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    moduleWeldPath = QtWidgets.QMainWindow()
    ui = Ui_moduleWeldPath()
    ui.setupUi(moduleWeldPath)
    moduleWeldPath.show()
    sys.exit(app.exec_())
